packages/gatsby/src/schema/node-model.js

```diff
@@ -81,9 +81,8 @@ class LocalNodeModel {
     this.schema = schema
     this.schemaComposer = schemaComposer
     this.createPageDependencyActionCreator = createPageDependency
-
-    this._rootNodeMap = _rootNodeMap
-    this._trackedRootNodes = _trackedRootNodes
+    this._rootNodeMap = _rootNodeMap || new WeakMap()
```

Comment:
the constructor is also called from `/gatsby/packages/gatsby/src/schema/context.ts` and the cache is passed there so that's why of the tests are failing. Not sure if it is ok to just create new `WeakMap/WeakSet` if they are not passed or the global cache can be share between the graphql-runner and the context

Reply:
Oh interesting. It seems generally a localNodeModal is passed in there (and perhaps always except for tests?) but yeah, this is a good fallback.